### PR TITLE
[docs] Document fontTools.merge

### PR DESCRIPTION
**Pull request by simoncozens**

### CODE REVIEW

File: Doc/source/merge.rst [hunk 1]
```diff
@@ -1,8 +1,10 @@
-#####
-merge
-#####
+####################################
+merge: Merge multiple fonts into one
+####################################
 
-.. automodule:: fontTools.merge
+``fontTools.merge`` provides both a library and a command line interface
+(``pyftmerge``) for merging multiple fonts together.
```
Comment (anthrotype) [inline Doc/source/merge.rst:6]: I would like eventually to get rid of the `pyft*` scripts and encourage people to use the `fonttools {subset,merge,...}` entry point. Maybe just mention `fonttools merge` here. We can decide to deprecate them for good

File: Lib/fontTools/merge.py [hunk 2]
```diff
@@ -941,6 +941,38 @@ def __delitem__(self, k):
...
+	This class merges multiple files into a single OpenType font, taking into
+	account complexities such as OpenType layout (``GSUB``/``GPOS``) tables and
+	cross-font metrics (e.g. ``hhea.ascent`` is set to the maximum value across
+	all the fonts).
+
+	If multiple glyphs map to the same Unicode value, subsequent glyphs are
+	renamed and a lookup in the ``locl`` feature will be created to disambiguate
+	them. For example, if the arguments are an Arabic font and a Latin font and
+	both contain a set of parentheses, the Latin glyphs will be renamed to
+	``parenleft#1`` and ``parenright#1``, and the equivalent of the following
```
Comment (anthrotype) [inline Lib/fontTools/merge.py:955]: The addition of hash characters to disambiguate glyph names in the merger reminds me of https://github.com/fonttools/fonttools/issues/1942

Unlike with that other issue (where a font contains duplicates in post table that fonttools needs to deduplicate internally and temporarily), in the case of the merger, we _are_ writing invalid glyph names (containing an invalid character in postscript `#`) in the merged font's `post` table.

Let's open a separate issue for this; maybe we can us a dot `.` here.

/cc @justvanrossum @jenskutilek

File: Lib/fontTools/merge.py [hunk 3]
```diff
@@ -941,6 +941,38 @@ def __delitem__(self, k):
...
+	feature code will be added:
+
+	.. code:: none
+
+		feature locl {
+			script latn;
+			language dflt;
+			sub [parenleft parenright] by [parenleft#1 parenright#1];
```
Comment (anthrotype) [inline Lib/fontTools/merge.py:963]: this snippet of FEA code is problematic given the `#` indicates inline comments (but I guess it's ok for the time being while we address that)

Comment (simoncozens) [reply Lib/fontTools/merge.py:963]: Thought of that. The `.. code:: none` turns off syntax highlighting for this block.

File: Lib/fontTools/merge.py (hunk 2 above)
Comment (khaledhosny) [reply Lib/fontTools/merge.py:955]: Note that there is also code that checks if the duplicate glyphs are the same, so there are more criteria considered than code point: https://github.com/fonttools/fonttools/blob/cfb261cf0279c352538f06f191c2df751477a177/Lib/fontTools/merge.py#L368-L377

File: Lib/fontTools/merge.py (hunk 3 above)
Comment (khaledhosny) [reply Lib/fontTools/merge.py:963]: I don’t find the code snippet to be adding much value IMHO, something like ”with a lookup mapping ``parenleft`` and ``parenright`` to ``parenleft#1`` and ``parenright#1``, respectively” would be enough.

Comment (khaledhosny) [reply Lib/fontTools/merge.py:963]: Also, if there is already a `locl` feature (with a script other than `DFLT`) the lookup will be inserted in it, so this needs to be mentioned. Also a `GSUB` table is required, merging a font with duplicate glyphs and no `GSUB` table raises an error.

Comment (simoncozens) [reply Lib/fontTools/merge.py:963]: I thought the choice of script/language for the locl feature was worth noting. We can say that the user can switch between the two glyphs using locl - but it's important to know what settings you need to use to get to which glyphset.

Comment (khaledhosny) [reply Lib/fontTools/merge.py:963]: I guess I have aversion to using FEA as _the_ OpenType language, and prefer tool-neutral description as we shouldn’t assume everyone reading the docs can readily understand this syntax.

Comment (simoncozens) [reply Lib/fontTools/merge.py:963]: OK, that's fair; I also agree that FEA is not the only language for feature code. (And certainly not the best ;-) ) Will rewrite it.

**Pull request merged**